packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -404,6 +378,106 @@ class IntellijValidator extends DoctorValidator {
   }
 }
 
+class IntellijValidatorOnLinux extends IntellijValidator {
+  IntellijValidatorOnLinux(String title, this.version, this.pluginsPath) : super(title);
+
+  @override
+  String version;
+
+  @override
+  String pluginsPath;
+
+  static Iterable<DoctorValidator> get installed {
+    List<DoctorValidator> validators = <DoctorValidator>[];
+    if (homeDirPath == null) return validators;
+    for (FileSystemEntity dir in new Directory(homeDirPath).listSync()) {
+      if (dir is Directory) {
+        String name = path.basename(dir.path);
+        IntellijValidator._idToTitle.forEach((String id, String title) {
+          if (name.startsWith('.$id')) {
+            String version = name.substring(id.length + 1);
+            String installPath;
+            try {
+              installPath = new File(path.join(dir.path, 'system', '.home')).readAsStringSync();
+            } catch (e) {
+              // ignored
+            }
+            if (installPath != null && FileSystemEntity.isDirectorySync(installPath)) {
+              String pluginsPath = path.join(dir.path, 'config', 'plugins');
+              validators.add(new IntellijValidatorOnLinux(title, version, pluginsPath));
+            }
+          }
+        });
+      }
+    }
+    return validators;
+  }
+}
+
+class IntellijValidatorOnMac extends IntellijValidator {
+  IntellijValidatorOnMac(String title, this.id, this.installPath) : super(title);
+
+  final String id;
+  final String installPath;
+
+  static final Map<String, String> _dirNameToId = <String, String>{
+    'IntelliJ IDEA' : 'IntelliJIdea',
+    'IntelliJ IDEA CE' : 'IdeaIC',
+    'WebStorm' : 'WebStorm',
+  };
+
+  static Iterable<DoctorValidator> get installed {
+    List<DoctorValidator> validators = <DoctorValidator>[];
+    for (FileSystemEntity dir in new Directory('/Applications').listSync()) {
+      if (dir is Directory) {
+        String name = path.basename(dir.path);
+        _dirNameToId.forEach((String dirName, String id) {
+          if (name == dirName) {
+            String title = IntellijValidator._idToTitle[id];
+            validators.add(new IntellijValidatorOnMac(title, id, dir.path));
+          }
+        });
+      }
+    }
+    return validators;
+  }
+
+  @override
+  String get version {
+    if (_version == null) {
+      String plist;
+      try {
+        plist = new File(path.join(installPath, 'Contents', 'Info.plist')).readAsStringSync();
+      } catch (e) {
+        // ignored
+      }
+      if (plist != null) {
+        int index = plist.indexOf('CFBundleShortVersionString');
```

Comment:
This could move inside the try-catch - you could avoid the `plist != null` check.